backend/strapi-sync/sync.js

```diff
@@ -0,0 +1,404 @@
+import { Op } from "sequelize";
+
+async function getPageData(url) {
+  try {
+    const response = await get(url);
+    const data = await response.json();
+
+    return data.data;
+  } catch (error) {
+    console.error(error);
+
+    return error;
+  }
+}
+
+export async function getData(url, queryParams) {
+  const items = [];
+
+  try {
+    let currentUrl = url;
+
+    if (queryParams) {
+      currentUrl = `${currentUrl}?${queryParams}`;
+    }
+
+    const response = await get(currentUrl);
+    const data = await response.json();
+    const pagination = data.meta.pagination;
+
+    items.push(...data.data);
+
+    let page = 1;
+
+    while (page < pagination.pageCount) {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      page += 1;
+
+      let paginatedUrl = currentUrl;
+
+      if (queryParams) {
+        paginatedUrl = `${paginatedUrl}&pagination[page]=${page}`;
+      } else {
+        paginatedUrl = `${paginatedUrl}?pagination[page]=${page}`;
+      }
+
+      const pageData = await getPageData(paginatedUrl);
+
+      items.push(...pageData);
+    }
+
+    return items;
+  } catch (error) {
+    console.error(error);
+    return [];
+  }
+}
+
+export async function fetchAllModels() {
+  const url = "https://cms.bcparks.ca/api";
+
+  const strapiData = [
+    {
+      endpoint: "/park-operations",
+      model: "park-operation",
+      fields: null,
+      items: [],
+    },
+    {
+      endpoint: "/park-operation-sub-areas",
+      model: "park-operation-sub-area",
+      fields: ["protectedArea", "parkSubAreaType"],
+      items: [],
+    },
+    {
+      endpoint: "/protected-areas",
+      model: "protected-area",
+      fields: ["parkFacilities", "parkOperations"],
+      items: [],
+    },
+    {
+      endpoint: "/camping-types",
+      model: "camping-type",
+      fields: null,
+      items: [],
+    },
+    {
+      endpoint: "/facility-types",
+      model: "facility-type",
+      fields: null,
+      items: [],
+    },
+    {
+      endpoint: "/park-operation-sub-area-types",
+      model: "park-operation-sub-area-type",
+      fields: ["facilityType", "campingType"],
+      items: [],
+    },
+    {
+      endpoint: "/park-operation-dates",
+      model: "park-operation-date",
+      fields: ["protectedArea"],
+      items: [],
+    },
+    {
+      endpoint: "/park-operation-sub-area-dates",
+      model: "park-operation-sub-area-date",
+      fields: ["parkOperationSubArea"],
+      items: [],
+    },
+  ];
+
+  await Promise.all(
+    strapiData.map(async (item) => {
+      const currentUrl = url + item.endpoint;
+      let qpString = null;
+
+      if (item.fields) {
+        const queryParams = [];
+
+        for (const field of item.fields) {
+          const param = `populate[${field}][fields]=id`;
+
+          queryParams.push(param);
+        }
+
+        qpString = queryParams.join("&");
+      }
+
+      const temp = getData(currentUrl, qpString);
+
+      item.items = temp;
```

Comment:
Can we just combine this onto one line?
```suggestion
      item.items = getData(currentUrl, qpString);
```